### PR TITLE
[IMP]bi_sql_editor. Select View Order.

### PR DESCRIPTION
**Pull request by aheficent**
Adds a new field that allows the user to select the view type.

### CODE REVIEW

File: bi_sql_editor/models/bi_sql_view.py [hunk 1]
```diff
@@ -30,6 +30,12 @@ class BiSQLView(models.Model):
         ('ui_valid', 'Graph, action and Menu Created'),
     ]
 
+    _VIEW_TYPE = [
```
Comment (legalsylvain) [inline bi_sql_editor/models/bi_sql_view.py]: Hi, 

1. due to your implementation, I think that the definition should better like this. 

    _VIEW_ORDER = [
        ('tree', 'tree / graph / pivot'),
        ('graph', 'Graph / Pivot / Tree'),
        ('pivot', 'Pivot / Graph / Tree'),
    ]

and view_order = fields.Selection.

2. Good implementation, but, I was thinking about possiilibity to disable a view, that is not possible with your implementation. 
https://github.com/OCA/reporting-engine/pull/148#issuecomment-316932609
(For exemple, In some cases, Tree view are really useless to display).

Comment (aheficent) [inline bi_sql_editor/models/bi_sql_view.py]: @legalsylvain  I think you're right. I'll set the field as a char then, and check that contains no errors

**Comment by aheficent** [comment]
@legalsylvain Changed the selection field by a char field as you proposed. Users can disable views now :relaxed:

File: bi_sql_editor/models/bi_sql_view.py [hunk 2]
```diff
@@ -59,6 +59,12 @@ class BiSQLView(models.Model):
 
     state = fields.Selection(selection_add=_STATE_SQL_EDITOR)
 
+    view_order = fields.Char(string='View Order',
+                             readonly=True,
+                             states={'draft': [('readonly', False)]},
```
Comment (legalsylvain) [inline bi_sql_editor/models/bi_sql_view.py]: you can set readonly to false if state in ['draft', 'sql_valid', 'model_valid'] .( well in fact if state != 'ui_valid')

File: bi_sql_editor/models/bi_sql_view.py [hunk 3]
```diff
@@ -59,6 +59,12 @@ class BiSQLView(models.Model):
...
+                             help='Comma-separated text. Possible values:'
+                                  ' "graph", "pivot" or "tree"')
+
```
Comment (legalsylvain) [inline bi_sql_editor/models/bi_sql_view.py]: Could you set a default value ? I guess 'pivot', 'graph', 'tree', by default, as this module is BI module and required=True.

File: bi_sql_editor/models/bi_sql_view.py [hunk 4]
```diff
@@ -357,12 +373,21 @@ def _prepare_search_view(self):
     @api.multi
     def _prepare_action(self):
         self.ensure_one()
+        view_mode = 'graph,pivot,tree'
+        view_id = self.graph_view_id.id
+        if self.view_order:
```
Comment (legalsylvain) [inline bi_sql_editor/models/bi_sql_view.py]: if view_order is required with default value, so this test is unnecessary;

File: bi_sql_editor/views/view_bi_sql_view.xml [hunk 5]
```diff
@@ -53,7 +53,8 @@ License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
                         <group>
                             <group>
                                 <field name="technical_name"/>
-                                <field name="view_name" />
+                                <field name="view_name"/>
+                                <field name="view_order" placeholder="e.g.:graph,pivot,tree"/>
```
Comment (legalsylvain) [inline bi_sql_editor/views/view_bi_sql_view.xml]: idem, placeholder become unnecessary with default value.

**Comment by aheficent** [comment]
Hi again :relaxed: I made the field required to avoid the double checks. Squashed the commits

**Comment by aheficent** [comment]
I just did a small fix  to change the contrains method name to _check_view_order

File: bi_sql_editor/models/bi_sql_view.py [hunk 6]
```diff
@@ -113,6 +121,16 @@ class BiSQLView(models.Model):
...
 
+    @api.constrains('view_order')
+    @api.multi
+    def _check_view_order(self):
+        for rec in self:
+            if rec.view_order:
+                for vtype in rec.view_order.split(','):
+                    if vtype not in ('graph', 'pivot', 'tree'):
+                        raise UserError(
+                            'Only graph, pivot or tree views are supported')
```
Comment (legalsylvain) [inline bi_sql_editor/models/bi_sql_view.py]: how, I didn't see but could you use _(), to allow translation ? 

regards.

Comment (aheficent) [inline bi_sql_editor/models/bi_sql_view.py]: Right! Done now

Comment (legalsylvain) [inline bi_sql_editor/models/bi_sql_view.py]: Thanks ! :+1:

**Pull request merged**